### PR TITLE
[Agent] Support invoking wasm_hook after parsing HTTP2 and gRPC

### PR DESCRIPTION
**Pull request by ZhuoZhuoCrayon**
### This PR is for:
- Agent

Support invoking wasm_hook after parsing HTTP2 and gRPC
- Extend functionality to invoke wasm_hook after successfully parsing HTTP2 and gRPC protocols
- Pass the parsed protocol type (e.g., HTTP, HTTP2, gRPC) to on_http_req and on_http_resp
- Previously, wasm_hook was only invoked after parsing HTTP, and wasm plugin OnHttpReq & OnHttpResp could not determine the parsed protocol type

Backport to branches
- v6.4

### CODE REVIEW

File: agent/src/flow_generator/protocol_logs/http.rs [hunk 1]
```diff
@@ -608,6 +612,7 @@ impl L7ProtocolParserInterface for HttpLog {
                 };
                 info.endpoint = Some(handle_endpoint(config, path));
             }
+            self.wasm_hook(param, payload, &mut info);
```
Comment (rvql) [inline agent/src/flow_generator/protocol_logs/http.rs:615]: `wasm_hook` should be moved before L604, before endpoint rewriting

Comment (rvql) [reply agent/src/flow_generator/protocol_logs/http.rs:615]: In uprobe mode (when `param.ebpf_type == EbpfType::GoHttp2Uprobe`), headers are reported in a way different from other modes: one payload contains one header. Calling wasm plugin on every payload would be wasted effort. We can skip calling plugins under this condition for now, or check if we have enough information (or headers) before calling plugins.

Comment (ZhuoZhuoCrayon) [reply agent/src/flow_generator/protocol_logs/http.rs:615]: > In uprobe mode (when `param.ebpf_type == EbpfType::GoHttp2Uprobe`), headers are reported in a way different from other modes: one payload contains one header. Calling wasm plugin on every payload would be wasted effort. We can skip calling plugins under this condition for now, or check if we have enough information (or headers) before calling plugins.

The `extra-log-fields` configuration is enough to collect header fields, while the wasm plugin is mainly used to parse the data. Therefore, in the case where param.ebpf_type == EbpfType::GoHttp2Uprobe, the call to the wasm plugin has been skipped.

**Comment by ZhuoZhuoCrayon** [comment]
@rvql PR updated

File: agent/src/flow_generator/protocol_logs/http.rs (hunk 1 above)
Comment (sharang) [reply agent/src/flow_generator/protocol_logs/http.rs]: When uprobe for grpc is enabled, kprobe for grpc will be disabled automatically.

So we should also handle uprobe case， but in this PR we can just leave a TODO and fix it in the future.

**Comment by rvql** [comment]
> @rvql PR updated

LGTM. Please help cherry-picking to `main` branch, thanks.

**Review by rvql — APPROVED** [review]

**Comment by ZhuoZhuoCrayon** [comment]
> > @rvql PR updated
> 
> LGTM. Please help cherry-picking to `main` branch, thanks.

thks!!!!, done 👉 #5903

**Pull request merged**